addons/account/models/chart_template.py

```diff
@@ -848,10 +848,18 @@ def _create_tax_templates_from_rates(self, company_id):
 
     @api.multi
     def existing_accounting(self, company_id):
-        model_to_check = ['account.move.line', 'account.invoice', 'account.payment', 'account.bank.statement']
-        for model in model_to_check:
-            if len(self.env[model].search([('company_id', '=', company_id.id)])) > 0:
```

Comment:
We will not have the same result with only adding limit=1 to the search ?
Not sure, but maybe it can even be faster, it will check model by model and it will stop if the first model satisfies the conditions. The SQL will test all the models.